tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go

```diff
@@ -199,182 +168,56 @@ func (r *RKE2NodeDriverProvisioningTestSuite) ProvisioningRKE2ClusterDynamicInpu
-
-		cloudCredential, err := provider.CloudCredFunc(client)
-		require.NoError(r.T(), err)
-		for _, kubeVersion := range r.kubernetesVersions {
-			name = tt.name + providerName + " Kubernetes version: " + kubeVersion
-			for _, cni := range r.cnis {
-				name += " cni: " + cni
-				r.Run(name, func() {
-					testSession := session.NewSession(r.T())
-					defer testSession.Cleanup()
-
-					testSessionClient, err := tt.client.WithSession(testSession)
-					require.NoError(r.T(), err)
-
-					clusterName := namegen.AppendRandomString(provider.Name)
-					generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
-					machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
-
-					machineConfigResp, err := testSessionClient.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
-					require.NoError(r.T(), err)
-
-					machinePools := machinepools.RKEMachinePoolSetup(tt.nodeRoles, machineConfigResp)
-
-					cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredential.ID, kubeVersion, machinePools)
+func (r *RKE2NodeDriverProvisioningTestSuite) testProvisioningRKE2Cluster(client *rancher.Client, provider Provider, nodesAndRoles []machinepools.NodeRoles, kubeVersion, cni string) {
+	cloudCredential, err := provider.CloudCredFunc(client)
 
-					clusterResp, err := clusters.CreateK3SRKE2Cluster(testSessionClient, cluster)
-					require.NoError(r.T(), err)
+	clusterName := namegen.AppendRandomString(provider.Name)
+	generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+	machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
 
-					kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
-					require.NoError(r.T(), err)
-
-					result, err := kubeProvisioningClient.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
-						FieldSelector:  "metadata.name=" + clusterName,
-						TimeoutSeconds: &defaults.WatchTimeoutSeconds,
-					})
-					require.NoError(r.T(), err)
-
-					checkFunc := clusters.IsProvisioningClusterReady
+	machineConfigResp, err := client.Steve.SteveType(provider.MachineConfigPoolResourceSteveType).Create(machinePoolConfig)
+	require.NoError(r.T(), err)
 
-					err = wait.WatchWait(result, checkFunc)
-					assert.NoError(r.T(), err)
-					assert.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
+	machinePools := machinepools.RKEMachinePoolSetup(nodesAndRoles, machineConfigResp)
 
-					clusterToken, err := clusters.CheckServiceAccountTokenSecret(client, clusterName)
-					require.NoError(r.T(), err)
-					assert.NotEmpty(r.T(), clusterToken)
+	cluster := clusters.NewK3SRKE2ClusterConfig(clusterName, namespace, cni, cloudCredential.ID, kubeVersion, machinePools)
 
-					clusterID, err := clusters.GetClusterIDByName(r.client, clusterName)
-					assert.NoError(r.T(), err)
-					podResults, podErrors := pods.StatusPods(client, clusterID)
-					assert.NotEmpty(r.T(), podResults)
```

Comment:
Since we're removing this test I still think we should add this check to the main provisioning test. Checking the status pods.

Reply:
That's a good idea, I went ahead and put it back here and in the 2.7 forwardport.